src/renderer/components/table/table-cell.tsx

```diff
@@ -20,9 +24,98 @@ export interface TableCellProps extends React.DOMAttributes<HTMLDivElement> {
   _sorting?: Partial<TableSortParams>; // <Table> sorting state, don't use this prop outside (!)
   _sort?(sortBy: TableSortBy): void; // <Table> sort function, don't use this prop outside (!)
   _nowrap?: boolean; // indicator, might come from parent <TableHead>, don't use this prop outside (!)
+
+  /**
+   * Triggers every time this cell is resized by user
+   * @param width desired width of this cell
+   */
+  _onResize?(width: number): void;
+
+  /**
+   * Triggers once the user has finished resizing this cell
+   */
+  _onResizeComplete?(): void; // triggers a callback when user stops resizing
 }
 
+type ResizeHandlerState = {
+  mousePosX?: number,
+};
+
 export class TableCell extends React.Component<TableCellProps> {
+  private resizeHandlerState?: ResizeHandlerState;
+  private cellContainer: React.RefObject<HTMLDivElement> = React.createRef();
+
+  /**
+   * Stores an instance of `<TableCell>` that is currently responsible for handling resize-related events
+   */
+  private static currentEventHandlingInstance?: TableCell;
+
+  /**
+   * Appends static event proxies to document `mousemove` and `mouseup` events for resizing,
+   * sets a particulate instance of `<TableCell>` to handle these events
+   * @param handlingInstance instance of `<TableCell>` responsible for handling events
+   */
+  private static addMouseEventListeners(handlingInstance: TableCell) {
+    TableCell.currentEventHandlingInstance = handlingInstance;
+
+    document.addEventListener("mousemove", TableCell.mouseMoveEventProxy);
+    document.addEventListener("mouseup", TableCell.mouseUpEventProxy);
+  }
```

Comment:
All this work has already been done with `src/renderer/components/resizing-anchor.tsx`. Is there a good reason not to reuse that?

Reply:
Wasn't really aware of this component's existence in our codebase, lol
Looking into it now